tests/compile-fail/non_extern_foreign_function_clashing.rs

```diff
@@ -0,0 +1,20 @@
+// revisions: two three
```

Comment:
I think this is the first time we are trying to use revisions in the Miri test suite, please test carefully if that actually works.

Reply:
I think it works well enough :tada:
```diff
diff --git a/tests/compile-fail/non_extern_foreign_function_clashing.rs b/tests/compile-fail/non_extern_foreign_function_clashing.rs
index 231b6f47..de10cd4e 100644
--- a/tests/compile-fail/non_extern_foreign_function_clashing.rs
+++ b/tests/compile-fail/non_extern_foreign_function_clashing.rs
@@ -15,6 +15,6 @@ fn main() {
         fn foo();
     }
     unsafe { foo() }
-    //[two]~^ ERROR unsupported operation: multiple definitions of `foo`
+    //[two]~^ ERROR unsupported operation: multiple definition of `foo`
     //[three]~^^ ERROR unsupported operation: multiple definitions of `foo`
 }
```
```
error in revision `two`: tests/compile-fail/non_extern_foreign_function_clashing.rs:17: unexpected error: '17:14: 17:19: unsupported operation: multiple definitions of `foo`'

error in revision `two`: tests/compile-fail/non_extern_foreign_function_clashing.rs:17: expected error not found: unsupported operation: multiple definition of `foo`

error in revision `two`: 1 unexpected errors found, 1 expected errors not found
status: exit status: 1
command: "/home/hyd-dev/Code/rust-lang/miri/target/x86_64-unknown-linux-gnu/release/miri" "tests/compile-fail/non_extern_foreign_function_clashing.rs" "-L" "/tmp/compiletestomgDnY" "--target=x86_64-unknown-linux-gnu" "--cfg" "two" "--error-format" "json" "-C" "prefer-dynamic" "-o" "/tmp/compiletestomgDnY/non_extern_foreign_function_clashing.stage-id" "-A" "unused" "--edition" "2018" "-Dwarnings" "-Dunused" "--sysroot" "/home/hyd-dev/.cache/miri/HOST" "-L" "/tmp/compiletestomgDnY/non_extern_foreign_function_clashing.stage-id.aux"
unexpected errors (from JSON output): [
    Error {
        line_num: 17,
        kind: Some(
            Error,
        ),
        msg: "17:14: 17:19: unsupported operation: multiple definitions of `foo`",
    },
]

not found errors (from test file): [
    Error {
        line_num: 17,
        kind: Some(
            Error,
        ),
        msg: "unsupported operation: multiple definition of `foo`",
    },
]

thread '[compile-fail] compile-fail/non_extern_foreign_function_clashing.rs' panicked at 'explicit panic', /home/hyd-dev/.cargo/registry/src/github.com-1ecc6299db9ec823/compiletest_rs-0.6.0/src/runtest.rs:1092:13
```
```diff
diff --git a/tests/compile-fail/non_extern_foreign_function_clashing.rs b/tests/compile-fail/non_extern_foreign_function_clashing.rs
index 231b6f47..5d6263ce 100644
--- a/tests/compile-fail/non_extern_foreign_function_clashing.rs
+++ b/tests/compile-fail/non_extern_foreign_function_clashing.rs
@@ -16,5 +16,5 @@ fn main() {
     }
     unsafe { foo() }
     //[two]~^ ERROR unsupported operation: multiple definitions of `foo`
-    //[three]~^^ ERROR unsupported operation: multiple definitions of `foo`
+    //[three]~^^ ERROR unsupported operation: multiple definition of `foo`
 }
```
```
error in revision `three`: tests/compile-fail/non_extern_foreign_function_clashing.rs:17: unexpected error: '17:14: 17:19: unsupported operation: multiple definitions of `foo`'

error in revision `three`: tests/compile-fail/non_extern_foreign_function_clashing.rs:17: expected error not found: unsupported operation: multiple definition of `foo`

error in revision `three`: 1 unexpected errors found, 1 expected errors not found
status: exit status: 1
command: "/home/hyd-dev/Code/rust-lang/miri/target/x86_64-unknown-linux-gnu/release/miri" "tests/compile-fail/non_extern_foreign_function_clashing.rs" "-L" "/tmp/compiletestMA1fFi" "--target=x86_64-unknown-linux-gnu" "--cfg" "three" "--error-format" "json" "-C" "prefer-dynamic" "-o" "/tmp/compiletestMA1fFi/non_extern_foreign_function_clashing.stage-id" "-A" "unused" "--edition" "2018" "-Dwarnings" "-Dunused" "--sysroot" "/home/hyd-dev/.cache/miri/HOST" "-L" "/tmp/compiletestMA1fFi/non_extern_foreign_function_clashing.stage-id.aux"
unexpected errors (from JSON output): [
    Error {
        line_num: 17,
        kind: Some(
            Error,
        ),
        msg: "17:14: 17:19: unsupported operation: multiple definitions of `foo`",
    },
]

not found errors (from test file): [
    Error {
        line_num: 17,
        kind: Some(
            Error,
        ),
        msg: "unsupported operation: multiple definition of `foo`",
    },
]

thread '[compile-fail] compile-fail/non_extern_foreign_function_clashing.rs' panicked at 'explicit panic', /home/hyd-dev/.cargo/registry/src/github.com-1ecc6299db9ec823/compiletest_rs-0.6.0/src/runtest.rs:1092:13
```